Source/WebKit/Platform/cocoa/CocoaHelpers.mm

```diff
@@ -188,7 +190,42 @@
     });
 }
 
-// MARK: NSDictionary helper methods.
+// MARK: JSON Helpers
+
+NSDictionary *parseJSON(NSData *json, NSError **error)
+{
+    if (!json)
+        return nil;
+    return dynamic_objc_cast<NSDictionary>([NSJSONSerialization JSONObjectWithData:json options:0 error:error]);
+}
+
+NSDictionary *parseJSON(NSString *json, NSError **error)
+{
+    return parseJSON([json dataUsingEncoding:NSUTF8StringEncoding], error);
+}
+
+NSDictionary *parseJSON(API::Data& json, NSError **error)
+{
+    return parseJSON(wrapper(json), error);
+}
+
+NSString *encodeJSONString(NSDictionary *dictionary, NSError **error)
+{
+    if (auto *data = encodeJSONData(dictionary, error))
+        return [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
```

Comment:
This file is ARC, it is safe to do.

Reply:
I believe this is safe (won't leak) because when the method is compiled under ARC (`CocoaHelpers.mm` is compiled with `-fobjc-arc`), ARC will use the naming convention (and lack of `NS_RETURNS_RETAINED` attribute) to autorelease the returned object regardless of whether the caller is ARC or non-ARC.